metaflow/plugins/kfp/kfp.py

```diff
@@ -1,316 +1,421 @@
+                    for parent_step in current_node.in_funcs:
+                        cur_input_path += "{parent}/{parent_task_id},".format(
+                            parent=parent_step,
+                            parent_task_id=str(step_to_task_id_map[parent_step]),
+                        )
+                    cur_input_path = cur_input_path.strip(",")
+                else:
+                    parent_step = current_node.in_funcs[0]
+                    cur_input_path = "{{run_id}}/{parent}/{parent_task_id}".format(
+                        parent=parent_step,
+                        parent_task_id=str(step_to_task_id_map[parent_step]),
+                    )
+
+            step_to_kfp_component_map[current_step] = build_kfp_component(
+                current_node, current_step, current_task_id, cur_input_path
+            )
+
+            for step in current_node.out_funcs:
+                if step not in seen_steps:
+                    steps_deque.append(step)
+                    seen_steps.add(step)
+
+        return step_to_kfp_component_map
+
+    def _step_cli(self, node, input_paths, user_code_retries, task_id):
+        """
+        Analogous to step_functions_cli.py
+        This returns the command line to run the internal Metaflow step click entrypiont.
+        """
+        from kfp import dsl
 
-    pipeline_func = create_kfp_pipeline_from_flow_graph(flow_graph, code_url)
-    run_pipeline_result = kfp.Client(namespace=api_namespace, userid=userid).create_run_from_pipeline_func(pipeline_func,
-                                                                     arguments={},
-                                                                     experiment_name=experiment_name,
-                                                                     run_name=run_name,
-                                                                     namespace=namespace)
-    return run_pipeline_result
+        cmds = []
 
+        script_name = os.path.basename(sys.argv[0])
+        executable = self.environment.executable(node.name)
 
-def create_kfp_pipeline_yaml(flow_graph, code_url, pipeline_file_path=DEFAULT_KFP_YAML_OUTPUT_PATH):
+        if R.use_r():
+            entrypoint = [R.entrypoint()]
+        else:
+            entrypoint = [executable, script_name]
+
+        paths = input_paths
+        kfp_run_id = "kfp-" + dsl.RUN_ID_PLACEHOLDER
+
+        # TODO: use dynamic task_id
+        #   But then we must save it in Dynamo (a kv store) for joins
+        #   task_id = "{{pod.name}}"  # ARGO pod name
+        if node.name == "start":
+            # We need a separate unique ID for the special _parameters task
+            task_id_params = "%s-params" % task_id
+            # Export user-defined parameters into runtime environment
+            param_file = "".join(
+                random.choice(string.ascii_lowercase) for _ in range(10)
+            )
+
+            # TODO: move to KFP plugin
```

Comment:
Safe to remove this todo? Looks like this is in the `kfp` plugin now :)

Reply:
This is referring to the `metaflow.plugins.aws.step_functions.set_batch_environment` step_functions command..

Reply:
Ah makes sense because you're potentially making an unnecessary extra subprocess call here? Instead if you can call it inline as a regular python function right?

Reply:
Not sure if it is unnecessary or not.  This is setting up some environment for Metaflow parameters.  Yun is working on the task to have KFP pipeline parameters work with these.